src/pytorch_lightning/loggers/csv_logs.py

```diff
@@ -146,7 +146,7 @@ def root_dir(self) -> str:
         If the experiment name parameter is an empty string, no experiment subdirectory is used and the checkpoint will
         be saved in "save_dir/version"
         """
-        return os.path.join(self.save_dir, self.name)
+        return os.path.join(str(self.save_dir), str(self.name))
```

Comment:
This casting seems redundant as `self.name` always has `str` typed in L214.
```suggestion
        return os.path.join(str(self.save_dir), self.name)
```

Reply:
Mypy is complaining if I don't pass strings.

Reply:
Are you sure? I don't get any error from mypy 0.950 locally.

Reply:
I think that when `self.name` is already declared as `str` (and not `Optional[str]`) thinks should work without explicit cast

Reply:
I have mypy 0.961. 

I'm working on setting up a common dev env at https://github.com/JustinGoheen/lightning/tree/dev/gitpod to help alleviate contributors having different dev envs.

Reply:
Whatever the CI jobs reports is meant to be the reference. I cannot reproduce the problem locally with the version you have.